core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -0,0 +1,124 @@
+package jenkins.install;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.logging.Logger;
+
+import javax.servlet.ServletContext;
+
+import org.jvnet.hudson.reactor.ReactorException;
+
+import hudson.WebAppMain;
+import hudson.init.InitStrategy;
+import hudson.model.Hudson;
+import hudson.security.GlobalSecurityConfigurationInclude;
+import hudson.security.SecurityRealm;
+import jenkins.I18n;
+import jenkins.model.AssetManager;
+import jenkins.model.Jenkins;
+
+/**
+ * A Jenkins instance used during first-run to provide a limited set of services while 
+ * initial installation is in progress
+ */
+public class SetupWizard extends Hudson {
```

Comment:
Is there a reason this doesn't extend `Jenkins`? If there is, then I think that reason should be clearly stated in the comments.